s3sync.go

```diff
@@ -368,7 +421,18 @@ func filterFilesForSync(sourceFileChan, destFileChan chan *fileInfo) chan *fileI
 			// 2. The dest doesn't have the same size as the source
 			// 3. The dest is older than the source
 			if !ok || sourceInfo.size != destInfo.size || sourceInfo.lastModified.After(destInfo.lastModified) {
-				c <- sourceInfo
+				c <- &fileOp{fileInfo: sourceInfo}
+			}
+			if ok {
+				destInfo.upToDate = true
```

Comment:
I think the flag name `upToDate` is a bit misleading because destInfo could have old `lastModified` here. I think the flag name should be something like `existsInSource`.